analyzer/restapi-plugin/src/main/java/eu/fasten/analyzer/restapiplugin/server/OpenAPIServer.java

```diff
@@ -0,0 +1,418 @@
+                 *  GET /api/mvn/au.org.consumerdatastandards:reflection/cg/1580706000
+                 */
+                routerFactory.addHandlerByOperationId
+                        ("get_call_graph_api__forge___pkg_name__cg__timestamp__get",
+                                routingContext -> {
+                                    RequestParameters params = routingContext.get("parsedParameters");
+                                    String forge = params.pathParameter("forge").getString();
+                                    String pkgName = params.pathParameter("pkg_name").getString();
+                                    long ts = params.pathParameter("timestamp").getInteger();
+                                    boolean transitive = params.queryParameter("transitive").getBoolean();
+                                    // TODO: check the transitive value and define it as false by default.
+
+                                    final var timestamp = this.getProperTimestamp(ts);
+
+                                    logger.debug("Parsed parameters converted to string (" + forge + ", "
+                                            + pkgName + ", " + timestamp + ").");
+
+                                    dslContext.connection(connection -> {
+                                        metadataDao.setContext(DSL.using(connection));
+                                        try {
+                                            String queryResult = metadataDao.getCallGraph(forge, pkgName, timestamp, transitive);
+                                            logger.debug("Got the reply: " + queryResult.substring(0, 50) + "...");
+
+                                            if (!queryResult.isEmpty())
+                                                routingContext.response().setStatusCode(200)
+                                                        .putHeader(HttpHeaders.CONTENT_TYPE, "application/json")
+                                                        .end(queryResult);
+                                            else
+                                                routingContext.fail(404, new Exception("Query not found!"));
+                                            // TODO: nice to have more specific exceptions ex: pkg not found, pkgName not found etc.
+                                        } catch (RuntimeException e) {
+                                            routingContext.fail(404, new Exception("No results found for this query!"));
+                                            logger.error("Error querying the database: ", e);
+                                            throw e;
+                                        }
+                                    });
+                                });
+
+                /**
+                 * "/api/{forge}/{pkg_name}/{version}"
+                 *
+                 * Given a package name and a version, retrieve all known metadata
```

Comment:
I thought we agreed to retrieve a metadata per database entity (i.e. `/api/{forge}/{pkg_name}/{version}/callables/metadata`, `/api/{forge}/{pkg_name}/{version}/edges/metadata` and etc)?

Reply:
Yes, in this case, the query was simplified to return the metadata for the package version, the other endpoints should be implemented as agreed in the last meeting.

Moreover, I agree with @gousiosg, this PR was intended to be a quick prototype showcasing canned queries as agreed in the Brussels meeting since the DB schema was still a work in progress.

To be able to move forward and implement a proper RESTful design we need clarification and the RESTful API Design Specification (deliverable 2.4)